middle_end/symbol_utils.ml

```diff
@@ -0,0 +1,55 @@
+(**************************************************************************)
+(*                                                                        *)
+(*                                 OCaml                                  *)
+(*                                                                        *)
+(*                       Pierre Chambart, OCamlPro                        *)
+(*           Mark Shinwell and Leo White, Jane Street Europe              *)
+(*                                                                        *)
+(*   Copyright 2013--2016 OCamlPro SAS                                    *)
+(*   Copyright 2014--2021 Jane Street Group LLC                           *)
+(*                                                                        *)
+(*   All rights reserved.  This file is distributed under the terms of    *)
+(*   the GNU Lesser General Public License version 2.1, with the          *)
+(*   special exception on linking described in the file LICENSE.          *)
+(*                                                                        *)
+(**************************************************************************)
+
+[@@@ocaml.warning "+a-9-30-40-41-42"]
+
+module CU = Compilation_unit
+
+let for_fun_ident ~compilation_unit loc id =
+  let compilation_unit =
+    match compilation_unit with
+    | None -> CU.get_current_exn ()
+    | Some cu -> cu
+  in
+  if Config.with_cpp_mangling then
+    (* CR lmaurer: Properly integrate [Compilation_unit.t] into [Mangling] *)
+    let unitname = CU.full_path_as_string compilation_unit in
+    let linkage_name =
+      Mangling.fun_symbol ~unitname ~loc ~id:(Ident.unique_name id)
+    in
+    Symbol.with_linkage_name compilation_unit linkage_name
+  else
+    Symbol.for_local_ident id
+
+module Flambda = struct
+  let for_variable var =
+    Symbol.for_name (Variable.get_compilation_unit var) (Variable.unique_name var)
+
+  let for_closure closure_id =
+    Symbol.for_name (Closure_id.get_compilation_unit closure_id)
+      (Closure_id.unique_name closure_id ^ "_closure")
+
+  let for_code_of_closure closure_id =
+    Symbol.for_name (Closure_id.get_compilation_unit closure_id)
+      (Closure_id.unique_name closure_id)
+
+  (* XXX Why is this needed?  Haven't we got the prefix correctly set now? *)
```

Comment:
I think we need to determine whether or not this is still needed.  As discussed my gut feeling is that it isn't, which would mean the import traversal in Flambda 1 could be removed amongst other things, but it needs checking.

Reply:
AFAIK, the only reason we wouldn't have a pack prefix right would be a missing .cmx, since the .cmi files don't yet have the pack prefix in the .cmi. Or I suppose `-no-alias-deps`.

Reply:
Should this be turned into a CR for phase 2 of this project then?

Reply:
Done.